include/nbl/builtin/hlsl/cpp_compat/vector.hlsl

```diff
@@ -73,6 +73,20 @@ NBL_TYPEDEF_VECTORS(float16_t);
 NBL_TYPEDEF_VECTORS(float32_t);
 NBL_TYPEDEF_VECTORS(float64_t);
 
+#ifndef __HLSL_VERSION
+template<typename T>
+concept is_vector_v =
```

Comment:
concepts are usually named just Camelcase without underscore, i.e. `Vector`

`is_vector_v` would be a template variable, i.e.
```cpp
template<typename T>
NBL_CONSTEXPR bool is_vector_v = ...;
```

Also there's a more sure way to make your trait work with partial specialization, which we ALREADY have!
https://github.com/Devsh-Graphics-Programming/Nabla/blob/c55d5a2e00e8a6271bdfd7b62ab6ffa8becefb97/include/nbl/builtin/hlsl/type_traits.hlsl#L613

So just add 
```cpp
template<typename T>
NBL_CONSTEXPR bool is_vector_v = is_vector<T>::value;
#ifndef __HLSL_VERSION
template<typename T>
concept Vector = is_vector_v<T>;
#endif
```
below `is_vector`